### DISCUSSION TITLE
Request for review of a Draft API for ridesharing/carpooling in MaaS apps

### DISCUSSION THREAD

**Discussion by gdelabie** [fabmob/standard-covoiturage#14] [Q&A]
This discussion thread is dedicated to a preliminary review of a standard for integration of ridesharing/carpooling services in MaaS apps/platforms. It is building on an existing standard RDEX and an API widely used in France for transport passenger information. The new « ridesharing API » will be first published in May (draft) with v1 final due by July (and based on the draft of May.
We’d like to receive feedbacks on the proposed design of API from stakeholders with experience in MaaS projects, most specifically ridesharing. This Github repo is the preferred place for discussion: this thread for general discussion / questions and issues for specific comments/questions to address.
[Current draft is available on a WIP branch](https://github.com/fabmob/standard-covoiturage/tree/WIP-operator-complete-api-initial-version). The main functions available are:
- Full planning information about carpooling rides available on the platform (though API queries)
- 2 booking options, one though API with full integration, the other through webURL (deep linking) with status tracking though the API
- Different options for using OpenIDConnect in the process or not

When publishing v1 of standard we will provide additional elements:
- Full documentation of the API
- Description of main use cases that can be covered, with best practices for implementation
- Analysis / comparison with TOMP-API for interoperability (TOMP-API has not been used because of an extensive use in France of an existing API with which we provide full retrocompatibility)
- Supporting tools : sandbox for testing interfaces, interoperability tools between the « ridesharing API » and TOMP-API

This discussion is primarily in English for documentation and international purposes. However comments/questions in French will be accepted.

**Comment by pierrecamilleri** [comment]
As the booking option through API with full integration and the return of booking information to the provider are still under discussion, they have not been merged yet to the main WIP branch. Until they are stabilized, the drafts can be found on the PRs #5 (booking information) and #6 (booking through API).

**Comment by levenes-mbc** [comment]
Hello, I was wondering if the authentification is mandatory for all search calls ?  Is the authentification related to the user or to the service?
For MaaS, we should have the possibility to indicate the journey to anonymous user, and then they authenticate to follow the process.

**Reply by gdelabie** [reply]
Hi Sylvain, authentification is not a prerequisite for search calls. It would be up to the MaaS app and transport operator to require authentification, but this is not specified in the standard.
The standard will allow provisions for OpenIDConnect-based authentication/user management for booking (but this will not be mandatory).
